### PR TITLE
MCKIN-13351 - Support for locked assets access by access token

### PR DESCRIPTION
**Pull request by naeem91**
This PR adds access token support to access LMS assets. This will enable mobile apps to access locked LMS assets by passing an access token as URL param instead of doing session authentication.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @naeem91! I've created [OSPR-4122](https://openedx.atlassian.net/browse/OSPR-4122) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: lms/envs/aws.py [hunk 1]
```diff
@@ -1183,3 +1183,8 @@
     'RATE_LIMIT_BACKEND_MAX_REQUESTS',
     RATE_LIMIT_BACKEND_MAX_REQUESTS
 )
+
+############### Settings for Assets access by using an access token ###############
+ASSETS_ACCESS_BY_TOKEN = ENV_TOKENS.get('ASSETS_ACCESS_BY_TOKEN', ASSETS_ACCESS_BY_TOKEN)
```
Comment (ihtram) [inline lms/envs/aws.py]: @naeem91 These tokens should be fetched from `AUTH_TOKENS` dict as `lms.auth.json` contains all the secret tokens. Please make desired changes in `mckinsey_internal` repo also

Comment (naeem91) [reply lms/envs/aws.py]: @ihtram Suggested changes made. I'll open a PR for internal and also request for placing actual key in secure repo.

Comment (ihtram) [reply lms/envs/aws.py]: @naeem91 Looks good now. I am approving this PR, please squash commits and open a PR for internal repo.

**Pull request merged**